apps/chat/src/components/Marketplace/TabRenderer.tsx

```diff
@@ -145,16 +146,24 @@ export const TabRenderer = ({ screenState }: TabRendererProps) => {
         : filteredEntities;
 
     const shouldSuggest =
-      selectedTab === MarketplaceTabs.MY_APPLICATIONS && !entitiesForTab.length;
+      selectedTab === MarketplaceTabs.MY_APPLICATIONS &&
+      !entitiesForTab.length &&
+      searchTerm.length;
 
     const groupedEntities = groupModelsAndSaveOrder(
       shouldSuggest ? filteredEntities : entitiesForTab,
     );
+
     const orderedEntities = groupedEntities.map(
       ({ entities }) => orderBy(entities, 'version', 'desc')[0],
     );
 
-    setSuggestedResults(shouldSuggest ? orderedEntities : null);
+    if (shouldSuggest) {
+      setSuggestedResults(orderedEntities);
+      return [];
+    } else {
+      setSuggestedResults([]);
+    }
```

Comment:
you shouldn't calculate `groupedEntities` and `orderedEntities` if you aren't going to use it
```suggestion
    if (shouldSuggest) {
      setSuggestedResults(orderedEntities);
      return [];
    }
    setSuggestedResults([]);
    const groupedEntities = groupModelsAndSaveOrder(
      shouldSuggest ? filteredEntities : entitiesForTab,
    );

    const orderedEntities = groupedEntities.map(
      ({ entities }) => orderBy(entities, 'version', 'desc')[0],
    );
    return orderedEntities ;
```

Reply:
`groupEntities` used to set suggested results as I see